receiver/tlscheckreceiver/config.go

```diff
@@ -42,77 +36,13 @@ type Config struct {
 	_ struct{}
 }
 
-func validatePort(port string) error {
-	portNum, err := strconv.Atoi(port)
-	if err != nil {
-		return fmt.Errorf("provided port is not a number: %s", port)
-	}
-	if portNum < 1 || portNum > 65535 {
-		return fmt.Errorf("provided port is out of valid range (1-65535): %d", portNum)
-	}
-	return nil
-}
-
 func validateTarget(ct *CertificateTarget) error {
-	// Check that exactly one of endpoint or file_path is specified
 	if ct.Endpoint != "" && ct.FilePath != "" {
 		return errors.New("cannot specify both endpoint and file_path")
 	}
 	if ct.Endpoint == "" && ct.FilePath == "" {
 		return errors.New("must specify either endpoint or file_path")
 	}
-
-	// Validate endpoint if specified
-	if ct.Endpoint != "" {
```

Comment:
Do these changes mean the collector can now start in an invalid state? We normally like to catch that at startup.

Reply:
We still do this validation, but we do it in the scrape function. The user can now include an invalid target, but it will be logged, the `otelcol_scraper_errored_metric_points` will be incremented, and the collector will not crash. As currently implemented, the collector will fail to start if there is an invalid target in the TLS Check configuration, such as a nonexistent file certificate. I do not think this is desirable behavior for the user. The new behavior matches that of the File Stats Receiver, HTTP Check Receiver, and others.